subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/release/TravisPlugin.java

```diff
@@ -52,27 +59,21 @@ public void apply(final Project project) {
         //set the branch to be checked out on ci build
         final GitCheckOutTask checkout = (GitCheckOutTask) project.getTasks().getByName(GitSetupPlugin.CHECKOUT_TASK);
         checkout.setRev(branch);
-        LazyConfiguration.lazyConfiguration(checkout, new Runnable() {
-            public void run() {
-                BasicValidator.notNull(checkout.getRev(),
-                        "Task " + checkout.getPath() + " does not know the target revision to check out.\n" +
-                        "In Travis CI builds, it is automatically configured from 'TRAVIS_BRANCH' environment variable.\n" +
-                        "If you are trying to run this task outside Travis, you can export the environment variable.\n" +
-                        "Alternatively, you can set the task's 'rev' property explicitly.");
-            }
-        });
+        LazyConfiguration.lazyConfiguration(checkout, () -> BasicValidator.notNull(checkout.getRev(),
+                "Task " + checkout.getPath() + " does not know the target revision to check out.\n" +
+                "In Travis CI builds, it is automatically configured from 'TRAVIS_BRANCH' environment variable.\n" +
+                "If you are trying to run this task outside Travis, you can export the environment variable.\n" +
+                "Alternatively, you can set the task's 'rev' property explicitly."));
 
         //update release needed task based on Travis' env variables
-        String pr = System.getenv("TRAVIS_PULL_REQUEST");
         LOG.info("Pull request from 'TRAVIS_PULL_REQUEST' env variable: {}", pr);
         final boolean isPullRequest = pr != null && !pr.trim().isEmpty() && !pr.equals("false");
         LOG.info("Pull request build: {}", isPullRequest);
 
-        project.getTasks().withType(ReleaseNeededTask.class, new Action<ReleaseNeededTask>() {
-            public void execute(ReleaseNeededTask t) {
-                t.setCommitMessage(System.getenv("TRAVIS_COMMIT_MESSAGE"));
-                t.setPullRequest(isPullRequest);
-            }
+        project.getTasks().withType(ReleaseNeededTask.class, t -> {
```

Comment:
We want to set the commit message on the GitCommitTask. We don't want to generate commit message for ReleaseNeededTask because this task should work on verbatim commit message supplied by Travis. ReleaseNeededTask is not making commits :)

Check out the initial design of this feature at #514

When you work on this feature, I suggest to test it this way:

```
./gradlew performRelease -PdryRun
```

Above will get you a local commit with the message. Please ensure that this commit has the stuff we need.

Thank you for working on this!!!

Reply:
Sure, I created PR to confirm my assumptions were right (which were wrong, but now I got feedback :)). 

I'll add necessary changes.